yamlgen/deploy/cert.yaml

```diff
@@ -1,36 +1,75 @@
 ---
 apiVersion: cert-manager.io/v1
-kind: ClusterIssuer
+kind: Issuer
 metadata:
   name: selfsigned-issuer
+  namespace: brupop-bottlerocket-aws
 spec:
   selfSigned: {}
 ---
 apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: brupop-selfsigned-ca
+  namespace: brupop-bottlerocket-aws
+spec:
+  isCA: true
+  commonName: brupop-selfsigned-ca
+  secretName: brupop-root-ca-secret
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS8
+  issuerRef:
+    name: selfsigned-issuer
+    kind: Issuer
+---
+apiVersion: cert-manager.io/v1
 kind: Issuer
 metadata:
-  name: my-ca-issuer
+  name: brupop-root-certificate-issuer
   namespace: brupop-bottlerocket-aws
 spec:
   ca:
-    secretName: brupop-tls
+    secretName: brupop-root-ca-secret
 ---
 apiVersion: cert-manager.io/v1
 kind: Certificate
 metadata:
   name: brupop-apiserver-certificate
   namespace: brupop-bottlerocket-aws
 spec:
-  isCA: true
-  commonName: my-selfsigned-ca
-  secretName: brupop-tls
+  secretName: brupop-apiserver-certificate
   privateKey:
     algorithm: RSA
     encoding: PKCS8
   dnsNames:
-    - brupop-apiserver.brupop-bottlerocket-aws.svc.cluster.local
-    - brupop-apiserver.brupop-bottlerocket-aws.svc
+    - "*.brupop-bottlerocket-aws.svc.cluster.local"
+    - "*.brupop-bottlerocket-aws.svc"
+  usages:
+    - server auth
+    - key encipherment
+    - digital signature
   issuerRef:
-    name: selfsigned-issuer
-    kind: ClusterIssuer
-    group: cert-manager.io
+    name: brupop-root-certificate-issuer
+    kind: Issuer
+---
+apiVersion: cert-manager.io/v1
+kind: Certificate
+metadata:
+  name: brupop-apiserver-client-certificate
+  namespace: brupop-bottlerocket-aws
+spec:
+  secretName: brupop-apiserver-client-certificate
+  privateKey:
+    algorithm: RSA
+    encoding: PKCS8
+  dnsNames:
+    - "*.brupop-bottlerocket-aws.svc.cluster.local"
+    - "*.brupop-bottlerocket-aws.svc"
+  usages:
+    - server auth
```

Comment:
Should this be `client auth` instead?

Reply:
We could then do something like [this](https://github.com/camelop/rust-mtls-example/blob/main/src/main.rs#L34) for the other bit -- using the client's keypair to establish mutual trust via the `identity()` method.

Reply:
Good call - yes that makes much more sense and will set us up well for mTLS